### PR TITLE
Add some definition files for waveshare board.

### PR DESCRIPTION
**Pull request by EngineerWill**
## Add some waveshare boards definition：
1. waveshare_rp2040_one
2. waveshare_rp2040_lcd_1.28
## Change waveshare_rp2040_lcd_0.96 default I2C
* Let waveshare board use the default I2C uniform

### CODE REVIEW

File: src/boards/include/boards/waveshare_rp2040_lcd_1.28.h [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+#endif
+#ifndef WAVESHARE_RP2040_LCD_RST_PIN
+#define WAVESHARE_RP2040_LCD_RST_PIN 12
+#endif
+#ifndef WAVESHARE_RP2040_LCD_BL_PIN
+#define WAVESHARE_RP2040_LCD_BL_PIN 25
+#endif
+// --- ADC ---
+#ifndef WAVESHARE_RP2040_BAT_ADC_PIN
+#define WAVESHARE_RP2040_LCD_BL_PIN 29
```
Comment (lurch) [inline src/boards/include/boards/waveshare_rp2040_lcd_1.28.h:84]: Typo? (These two lines don't match)

File: src/boards/include/boards/waveshare_rp2040_lcd_0.96.h [hunk 2]
```diff
@@ -31,13 +31,13 @@
 
 // --- I2C ---
 #ifndef PICO_DEFAULT_I2C
-#define PICO_DEFAULT_I2C 0
+#define PICO_DEFAULT_I2C 1
 #endif
 #ifndef PICO_DEFAULT_I2C_SDA_PIN
-#define PICO_DEFAULT_I2C_SDA_PIN 4
+#define PICO_DEFAULT_I2C_SDA_PIN 6
```
Comment (lurch) [inline src/boards/include/boards/waveshare_rp2040_lcd_0.96.h:37]: How come this has changed? Was the original _always_ wrong, or are there now two separate revisions of this board?

File: src/boards/include/boards/waveshare_rp2040_lcd_1.28.h (hunk 1 above)
Comment (EngineerWill) [reply src/boards/include/boards/waveshare_rp2040_lcd_1.28.h]: Yes, it was my fault. Has been revised.

File: src/boards/include/boards/waveshare_rp2040_lcd_0.96.h (hunk 2 above)
Comment (EngineerWill) [reply src/boards/include/boards/waveshare_rp2040_lcd_0.96.h:37]: The board is not modified, and it is not an wrong, but Waveshare's extension board usually uses I2C pins are GPIO6 and GPIO7, modify the I2C description here to help users use

**Review by kilograham — APPROVED** [review]

**Comment by tomoveu** [comment]
@EngineerWill I just received 5 units of RP2040_lcd_1.24 and they all are B2 revision. Are yours B1?

@kilograham could you please add information in the README that new features first land in branch develop?

Today, I received my boards and made a [RP2040-1.24 board definition](https://github.com/tomoveu/pico-sdk/tree/add-rp2040-lcd-1-28), and only after I opened a PR, I got the info that "PR must be against develop". Imagine if I have worked on something larger and then realized my work is a duplicate. 

Thank you both for continuously improving the SDK :100: I hope this very neat new PR2040 board makes the next release :)

**Comment by kilograham** [comment]
> @EngineerWill I just received 5 units of RP2040_lcd_1.24 and they all are B2 revision. Are yours B1?
> 
> @kilograham could you please add information in the README that new features first land in branch develop?
> 
> Today, I received my boards and made a [RP2040-1.24 board definition](https://github.com/tomoveu/pico-sdk/tree/add-rp2040-lcd-1-28), and only after I opened a PR, I got the info that "PR must be against develop". Imagine if I have worked on something larger and then realized my work is a duplicate.
> 
> Thank you both for continuously improving the SDK 💯 I hope this very neat new PR2040 board makes the next release :)

good point; have done so.

**Comment by EngineerWill** [comment]
> @EngineerWill I just received 5 units of RP2040_lcd_1.24 and they all are B2 revision. Are yours B1?
> 
> @kilograham could you please add information in the README that new features first land in branch develop?
> 
> Today, I received my boards and made a [RP2040-1.24 board definition](https://github.com/tomoveu/pico-sdk/tree/add-rp2040-lcd-1-28), and only after I opened a PR, I got the info that "PR must be against develop". Imagine if I have worked on something larger and then realized my work is a duplicate.
> 
> Thank you both for continuously improving the SDK 💯 I hope this very neat new PR2040 board makes the next release :)

```
// All boards have B1 RP2040
#ifndef PICO_RP2040_B0_SUPPORTED 
#define PICO_RP2040_B0_SUPPORTED  0
#endif
```
* Are you confused about the remarks?
  *  The old remarks are used here
  *  but the actual expression is that the B0 version is not supported
  *  It won't affect the product you have on hand

**Comment by tomoveu** [comment]
@EngineerWill How to make sure that my `SDK build` is for the `B2 revision`? 

```
// All boards have B1 RP2040
```

Yes, this text made me wonder because all five boards came with `B2 revision`. Please see the photo below:

Image

**Comment by EngineerWill** [comment]
> @EngineerWill How to make sure that my `SDK build` is for the `B2 revision`?
> 
> ```
> // All boards have B1 RP2040
> ```
> 
> Yes, this text made me wonder because all five boards came with `B2 revision`. Please see the photo below:
> 
> Image


It seems that the B0 version was only in circulation in the early days.
There is no difference between B1 and B2 in SDK. 
The specific difference may be that B2 fixes some bugs

**Comment by tomoveu** [comment]
@EngineerWill Thank you for this information :)

**Pull request merged**